Finished Modus styling (and tweaked a bunch of other things while at it)

Eneroth3 opened the pull request:
- [x] Squash

Eneroth3 commented:
@thomthom, for some reason I can't add you as a reviewer

thomthom commented on src/step07.rb:
```diff
@@ -153,8 +154,10 @@ def self.texture_to_hash(texture)
-      model_height_formatted: texture.height.to_l.to_s,
+      # Replace spaces with non-breaking spaces in lengths to avoid
+      # "~" sign or unit from being on a different line tha the number.
+      model_width_formatted: texture.width.to_l.to_s.gsub(" ", "\u00A0"),
```
Instead of manipulating the strings - I'd think we can use CSS to control this. (This feels like changing the source text to all capitals when you can instead apply a `text-transform`)

https://developer.mozilla.org/en-US/docs/Web/CSS/white-space
https://developer.mozilla.org/en-US/docs/Web/CSS/word-break

Might need to wrap a span around the text where we don't want the lines to break.

thomthom replied:
`tha the` ?

thomthom commented on src/html/step07.html:
```diff
@@ -208,7 +204,8 @@ <h2>[No Material Selected]</h2>
         },
         texturePreview: function() {
-          // Forces the browser to reload the image.
+          // Appending a dummy query parameter rorces the browser to reload the
```
`rorces` => `forces`

thomthom commented on src/html/step07.html:
```diff
@@ -1,23 +1,20 @@
 <head>
-  <link href="../vendor/Semantic-UI/semantic.min.css" rel="stylesheet">
-  <script src="../vendor/jquery.js"></script>
-  <script src="../vendor/Semantic-UI/semantic.min.js"></script>
```
The `Semantic-UI/` folder is completely unused now, right? We can nuke it?

thomthom commented on src/vendor/jquery.js:
```diff
@@ -1,4 +0,0 @@
-/*! jQuery v3.2.1 | (c) JS Foundation and other contributors | jquery.org/license */
```
We don't need jQuery for anything? Modus is built on Bootstrap, which in turn uses jQuery.

thomthom replied:
Oh I see - it's now part of the Modus subfolder. Never mind me.

thomthom commented:
Any screenshots of how the dialogs now look?

thomthom commented on src/step07.rb:
```diff
@@ -9,38 +9,39 @@ module Step07
+      :resizable => true,
+      :width => width,
+      :height => 500,
+      :use_content_size => true
```
Should we account for older SU versions that might need some extra space when content size is not set?

Eneroth3 replied:
Good catch. Adding some extra space.

thomthom commented on src/step07.rb:
```diff
@@ -9,38 +9,42 @@ module Step07
+    width = 15 + 128 + 15 + 15 + 128 + 15 # Margins and contents
+    # Add extra space for older SU versions that doesn't support control of
+    # inner size.
+    width += 20 if Sketchup.version.to_f < 21.1
```
Only the width? The titlebar adds more than the size of the frame.

thomthom commented on README.md:
```diff
@@ -1,17 +1,21 @@
+* [Communication between Ruby and JS.](#part-2-communication-between-ruby-and-js) HTML content synchronization.
+* [Styling.](#part-3-styling)
 
 ![](Screenshot.png)
```
From the screenshot there seems to be rather much excess space between the dialog content an the footer buttons. Maybe shrink that a bit?

Eneroth3 replied:
Fixed height and updated image.

thomthom commented on src/step02.rb:
```diff
@@ -6,6 +6,9 @@ module Step02
 
   def self.create_dialog
+    # The content and callbacks are lost when the dialog is closed.
+    # The next time you show the dialog it's empty and broken.
```
I think we should reword this. "lost", "broken" etc.

Also, I don't recall the content being gone, it's only the action callbacks that needs to be reinitialized. (However this dialog might rely on the callbacks to populate the content?)

How about?

```
# The callbacks needs to be reinitialized after the dialog has been closed.
```

Eneroth3 replied:
Oh, I thought the HTML needed to be re-added too. Changing this.

I think "reattached" nicely as it matches the method name.

thomthom commented on src/step07.rb:
```diff
@@ -9,18 +9,22 @@ module Step07
+    # Define pixel perfect dimensions matching our content.
+    width = 15 + 128 + 15 + 15 + 128 + 15
+    height = 460
+    # Add some extra space for older SSU versions that don't support inner sizing.
```
"SSU" => "SU"

Eneroth3 replied:
You have a problem with Sveriges socialdemokratiska ungfomsförbund?

(Fixed it)

thomthom commented:
Final ask: Can we adjust the padding ah the bottom of 07 example?

https://github.com/SketchUp/htmldialog-examples/pull/4/files?short_path=67b6a55#diff-67b6a55e5b287dc1d9280dc1f32cb4bd25d966ebc8f589431f28445d311631b3

Padding is much thinner than around the top and edges. Looks very bottom heavy.

Eneroth3 commented:
Testing.

@thomthom, can you respond to this?

Eneroth3 commented:
Fixed bottom button bottom padding!

Eneroth3 committed:
Switch to Trimble Modus design system

Includes other minor UX tweaking such as clarifying dialog button text, clarifying toolbar button titles and adding toolbar separators.

thomthom commented:
> Testing.
> 
> @thomthom, can you respond to this?

Only by using the comment field at the bottom here. From what I understand, comments made here are part of the Conversation tab. And these comments aren't individually threaded.

What makes it confusing (IMO) is that we see the code comments along side.

Makes me feel it's better to isolate conversations via code comments.

Comments here can be quoted via the kebab menu:
![image](https://user-images.githubusercontent.com/192418/107012213-05ce5e80-6799-11eb-9182-803d91cbcaab.png)

thomthom reviewed (approved):
Shiny!
